cmd/api/src/database/assetgrouptags.go

```diff
@@ -408,3 +445,78 @@ func (s *BloodhoundDB) GetSelectorNodesBySelectorIds(ctx context.Context, select
 	}
 	return nodes, CheckError(s.db.WithContext(ctx).Raw(fmt.Sprintf("SELECT selector_id, node_id, certified, certified_by, source, created_at, updated_at FROM %s WHERE selector_id IN ?", model.AssetGroupSelectorNode{}.TableName()), selectorIds).Find(&nodes))
 }
+
+func (s *BloodhoundDB) GetMaxTierPosition(ctx context.Context) (int32, error) {
+	var max null.Int32
+	var tag model.AssetGroupTag
+
+	if result := s.db.WithContext(ctx).
+		Raw(fmt.Sprintf("SELECT MAX(position) FROM %s", tag.TableName())).
+		Scan(&max); result.Error != nil {
+		return 0, CheckError(result)
+	}
+
+	if !max.Valid {
+		// no rows in table
+		return 0, nil
+	}
+
+	return max.Int32, nil
```

Comment:
This might be easier
```suggestion
	return max.ValueOrZero(), nil
```